ksql-api/src/test/java/io/confluent/ksql/api/ApiTest.java

```diff
@@ -391,52 +469,57 @@ public void shouldHandleUnknownQueryIDInCloseQuery() throws Exception {
   @Test
   public void shouldInsertWithNoAcksStream() throws Exception {
 
+    // Given
     JsonObject params = new JsonObject().put("target", "test-stream").put("requiresAcks", false);
-
     List<JsonObject> rows = generateInsertRows();
     Buffer requestBody = Buffer.buffer();
     requestBody.appendBuffer(params.toBuffer()).appendString("\n");
     for (JsonObject row : rows) {
       requestBody.appendBuffer(row.toBuffer()).appendString("\n");
     }
 
+    //When
     HttpResponse<Buffer> response = sendRequest("/inserts-stream", requestBody);
-    assertEquals(200, response.statusCode());
-    assertEquals("OK", response.statusMessage());
 
+    // Then
+    assertThat(response.statusCode(), is(200));
+    assertThat(response.statusMessage(), is("OK"));
     waitUntil(() -> rows.equals(testEndpoints.getInsertsSubscriber().getRowsInserted()));
-    assertTrue(testEndpoints.getInsertsSubscriber().isCompleted());
-    assertEquals("test-stream", testEndpoints.getLastTarget());
+    assertThatEventually(() -> testEndpoints.getInsertsSubscriber().getRowsInserted(), is(rows));
```

Comment:
Did you mean to remove the `waitUtil` row above? It looks to test the same as this new line...